pkg/controller/job/job_controller.go

```diff
@@ -975,6 +975,27 @@ func (jm *Controller) syncJob(ctx context.Context, key string) (rErr error) {
 	return manageJobErr
 }
 
+func (jm *Controller) newFailureCondition(reason, message string) *batch.JobCondition {
+	cType := batch.JobFailed
+	if delayTerminalCondition() {
+		cType = batch.JobFailureTarget
+	}
+	return newCondition(cType, v1.ConditionTrue, reason, message, jm.clock.Now())
+}
+
+func (jm *Controller) newSuccessCondition() *batch.JobCondition {
+	cType := batch.JobComplete
+	if delayTerminalCondition() {
+		cType = batch.JobSuccessCriteriaMet
+	}
+	return newCondition(cType, v1.ConditionTrue, "", "", jm.clock.Now())
```

Comment:
I'm sorry if I am missing any context and background.

I'm wondering if we should have a dedicated reason for these situations where the job doesn't have successPolicy, and the `deplayTerminalConfiton` is true. Because I guess that users can not understand the reason why job has `SuccessCriteriaMet` regardless of the job doesn't have any successPolicies.

What do you think?

Reply:
I'm +1 on a dedicated reason for completing a job when `spec.completions == status.succeeded`, one natural name to consider would be `CompletionsReached`.

However. we didn't have the reason prior to this PR, so I don't consider it a regression (see [here[(https://github.com/kubernetes/kubernetes/blob/a39f42582f065de4b4708deef6709229f5c4d49b/pkg/controller/job/job_controller.go#L928)). I think we could add it as part of the SuccessPolicy KEP since it is still alpha / beta, but it would be a bit of an API-breaking change. WDYT @alculquicondor @soltysh @deads2k ?

Reply:
I think it's safe and reasonable to add a reason explaining where the success criteria is coming from. I wouldn't consider that a breaking change, and can be easily addressed in followup or as part of the other work, as Michal mentioned.

Reply:
That sounds reasonable to me. Thanks @mimowo @soltysh!
Actually, I'm working on the SuccessPolicy graduation PR. So, let me try it on the SuccessPolicy beta graduation PR. If we find any drawback to the dedicated reason in the graduation PR, we can give up to introduce the dedicated reason.